File: week1/Evaluate Reverse Polish Notation.js
var evalRPN = function (tokens) {
	return evaluate(tokens);
};

function evaluate(tokens) {
	const stack = [];
	for (let token of tokens) {
		if (isOperator(token)) {
			const operator = token;
			const secondOperand = stack.pop();
			const firstOperand = stack.pop();

			const result = operate(operator, firstOperand, secondOperand);
			stack.push(result);
		} else {
			stack.push(token);
		}
	}

	return stack[0];
}

function isOperator(token) {
	return token === "+" || token === "-" || token === "*" || token === "/";
}

function operate(operand, num1, num2) {
	//convert to integers
	num1 = +num1;
	num2 = +num2;

	//operate
	if (operand === "+") return num1 + num2;
	else if (operand === "-") return num1 - num2;
	else if (operand === "*") return num1 * num2;
	else if (operand === "/") return Math.trunc(num1 / num2);
}

File: week10-after-admission/Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        num_stack = []
        
        for token in tokens:
            if token.isnumeric():
                num_stack.append(int(token))
            else:
                num2 = num_stack.pop() 
                num1 = num_stack.pop() 
                
                result = None
                if token == "+":
                    result = num1 + num2
                elif token == "-":
                    result = num1 - num2
                elif token == "*":
                    result = num1 * num2
                elif token == "/":
                    result == num1//num2
                
                num_stack.append(result)
                
        return num_stack.pop()
    
#15min                    
#1sub
                
            
        
